### PR TITLE
Warn about and ignore empty local/param names in name section

### PR DESCRIPTION
**Pull request by kripken**
Fixes the crash in #4418

Also replace the `.at()` there with better logic to handle imported functions.

### CODE REVIEW

**Comment by timkendrick** [comment]
Thanks for the speedy fix! I've compiled this version and can confirm that it allows modules containing imported functions to compile successfully with the `--debuginfo` argument.

As explained in [my comment on the original issue](https://github.com/WebAssembly/binaryen/issues/4418#issuecomment-1006402970) I think it's probably the wrong behavior for this to emit a warning seeing as as far as I understand it the WASM binary is perfectly legitimate - I'll leave that to you guys though; from my perspective I'm fine with spurious warnings so long as it compiles.

Thanks again for your help on this, and for all your work on `wasm-opt`, it's a lifesaver!

File: test/lit/binary/empty-param-name.test [hunk 1]
```diff
@@ -0,0 +1,6 @@
+;; Test that we show a warning on an empty local name, and do not crash.
```
Comment (aheejin) [inline test/lit/binary/empty-param-name.test:1]: - Would be good to include the contents of the wasm file
- I think this depends on the wabt's behavior that emits an empty local name? Would it be good to reproduce the behavior from our end?

Comment (kripken) [reply test/lit/binary/empty-param-name.test:1]: Hmm, I consider wabt's behavior to be a bug. Maybe not literally, it's valid with the spec, but an empty name is just worse for debugging than an index. So it's better to not emit it. I opened a wabt issue meanwhile: https://github.com/WebAssembly/wabt/issues/1799

But I guess it's debatable. I'm ok with removing the warning (and skipping such names so we don't crash).

Comment (kripken) [reply test/lit/binary/empty-param-name.test:1]: (added content details to the testcase as requested, thanks)

Comment (aheejin) [reply test/lit/binary/empty-param-name.test:1]: If it's a bug from the wabt side, isn't this test relying on that bug? Would this test still work after that bug is fixed?

Comment (kripken) [reply test/lit/binary/empty-param-name.test:1]: Yes, if wabt fixes the bug then recreating the wasm won't cause a warning. Still, I think it's good to test that we don't crash + that we warn on an invalid input in a clear way. I don't feel strongly though.

I also am ok with either warning or not warning.

Comment (aheejin) [reply test/lit/binary/empty-param-name.test:1]: Ah right, we still have the binary.. I was confused and thought if the wabt bug is fixed this test would no longer work, sorry. This sounds OK, especially given that the test file description says the file was generated from the bug.

**Review by aheejin — APPROVED** [review]

**Comment by kripken** [comment]
I'll land this with the warning for now. It it turns out annoying, or discussion on the wabt issue decides this isn't a wabt bug, we can always remove it.

**Comment by timkendrick** [comment]
Thanks @kripken, makes sense. Thanks for all your help fixing this!

**Pull request merged**